content/blog/2020/05/2020-05-07-docker-windows-agents.adoc

```diff
@@ -0,0 +1,116 @@
+* `-Secret <SECRET>` or JENKINS_SECRET - the secret as shown on the master after creating the agent
+* `-Name <NAME>` or JENKINS_AGENT_NAME - the name of the agent, it should match the name you specified when creating the agent on the master
+* `-Tunnel <HOST:PORT>` or JENKINS_TUNNEL - A tunnel host to route the TCP traffic through when connecting to the master
+* `-WorkDir <WORKDIR>` or JENKINS_AGENT_WORKDIR - same as the -workDir parameter mentioned above in the jenkins/agent image information.
+* `-WebSocket` or JENKINS_WEB_SOCKET=true - when present the connection to the master will be done via WebSocket instead of TCP networking.
+* `-DirectConnection <HOST:PORT>` or JENKINS_DIRECT_CONNECTION - Connect directly to this TCP agent port, skipping the HTTP(S) connection parameter download.
+* `-InstanceIdentity <BASE64VALUE>` or JENKINS_INSTANCE_IDENTITY - The base64 encoded InstanceIdentity byte array of the Jenkins master. When this is set, the agent skips connecting to an HTTP(S) port for connection info
+* `-Protocols <PROTOCOLS>` or JENKINS_PROTOCOLS - Specify the remoting protocols to attempt when instanceIdentity is provided
+* `-JavaHome <JAVA_HOME>` or JAVA_HOME  - An override for the default JAVA_HOME baked into the image (different for JDK8 vs. JDK11)
+
+Example:
+
+```
+docker run jenkins/inbound-agent:windowsservercore-1809 -Url http://jenkins-server:port -WorkDir=C:/Users/jenkins/Agent -Secret <SECRET> -Name <AGENTNAME>
+```
+
+Example using environment variables:
+
+```
+docker run -e "JENKINS_URL=http://jenkins-server:port" -e "JENKINS_AGENT_NAME=AGENTNAME" jenkins/inbound-agent:windowsservercore-1809 -WorkDir=C:/Users/jenkins/Agent -Secret <SECRET> -Name <AGENTNAME>
+```
+
+The URL, Name and Secret parameters are required, but can be specified as either command line parameters or environment variables.
+
+
+
+=== jenkins/ssh-agent
+
+As mentioned above the jenkins/ssh-agent docker image is based on SSH communication with the master, rather than the remoting TCP or WebSocket protocols. This communication is initiated by the master rather than the agent connecting to the master is is done with jenkins/agent and jenkins/inbound-agent.
+
+The image sets up a `jenkins` user and the OpenSSH server so that the master can connect to the agent via SSH. The image expects an SSH public key as a parameter and puts that key into the authorized_keys file for the jenkins user. The private key should be specified in the agent configuration on the master to allow the master to connect correctly.
+
+Example:
+
+```
+docker run jenkins/ssh-agent "<public key>"
+```
+
+You can also pass the public key as an environment variable when using `docker run`.
+
+Example:
+
+```
+docker run -e "JENKINS_AGENT_SSH_PUBKEY=<public key>" jenkins/ssh-agent
+```
+
+You will then be able to connect this agent using the link:plugins.jenkins.io/ssh-slaves/[SSH Build Agents plugin] as "jenkins" with the matching private key.
```

Comment:
Isn’t there a plugin macro that can be used so you don’t need the full url?

Reply:
Thanks, I didn't know that was there.